utils/backoff.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/cenkalti/backoff/v4"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// WithMaxRetriesLog runs the operation until it succeeds or max retries has been reached.
+// It uses exponential back off.
+// It optionally logs information if logger is set.
+func WithMaxRetriesLog(
```

Comment:
Rather than separate with/without log functions, I think we should instead have a single method that takes a `logging.Logger` and emits warning logs on retries. I can't think of any use cases where we'd want retry attempts to be logged as Warn for some operations, but not logged at all for others.